### PR TITLE
chore: upgrade to .net core 7.0.16

### PR DESCRIPTION
**Pull request by hkfgo**
Fixes #

### CODE REVIEW

File: src/output.txt [hunk 1]
```diff
@@ -0,0 +1,283 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (tomkerkhove) [inline src/output.txt:1]: Can we delete this file?

File: src/Promitor.Agents.ResourceDiscovery/Dockerfile.windows [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/dotnet/sdk:7.0.402 AS build
+FROM mcr.microsoft.com/dotnet/sdk:7.0.406 AS build
```
Comment (tomkerkhove) [inline src/Promitor.Agents.ResourceDiscovery/Dockerfile.windows:1]: Can you update the Linux image as well please?

Comment (hkfgo) [reply src/Promitor.Agents.ResourceDiscovery/Dockerfile.windows:1]: Forgot to commit it sorry :)

Comment (tomkerkhove) [reply src/Promitor.Agents.ResourceDiscovery/Dockerfile.windows:1]: No worries!

**Comment by hkfgo** [comment]
Do you mind re-running that pipeline again? Unless I can do it too even as a non-maintainer? Integration tests seem to be just hanging without any obvious failures.

**Comment by hkfgo** [comment]
On a separate note, do you think this [PR](https://github.com/tomkerkhove/promitor/pull/2410) to bump AzureIdentity looks good? We are hoping for release that includes both PRs to address CVEs.

**Comment by tomkerkhove** [comment]
It looks like the CI is timing out everywhere for some reason so I'm checking that.

In meantime, I'll likely upgrade to .NET 8 given that is GA now: https://github.com/tomkerkhove/promitor/pull/2402

**Comment by hkfgo** [comment]
>It looks like the CI is timing out everywhere for some reason so I'm checking that.

Thanks. 

>In meantime, I'll likely upgrade to .NET 8 given that is GA now: https://github.com/tomkerkhove/promitor/pull/2402
I was going to take that on actually. Do you mind if I go ahead with that? Just for my own learning mostly.

**Pull request closed without merging**